packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -64,19 +68,29 @@ export class DetachedFieldIndex {
 			this.revisionTagCodec,
 			this.options,
 		);
-		populateNestedMap(this.detachedNodeToField, clone.detachedNodeToField, true);
+		// populate the rangeMap of detached nodes
+		for (const [major, innerRangeMap] of this.detachedNodeRangeMap) {
+			const clonedRangeMap: RangeMap<ForestRootId> = [];
+			for (const rangeEntry of innerRangeMap) {
+				clonedRangeMap.push({ ...rangeEntry });
+			}
+			clone.detachedNodeRangeMap.set(major, clonedRangeMap);
+		}
 		return clone;
 	}
 
+	/**
+	 * Returns the atomic detached node id instead of ranges
+	 */
 	public *entries(): Generator<{ root: ForestRootId } & { id: Delta.DetachedNodeId }> {
-		for (const [major, innerMap] of this.detachedNodeToField) {
-			if (major !== undefined) {
-				for (const [minor, entry] of innerMap) {
-					yield { id: { major, minor }, root: entry };
-				}
-			} else {
-				for (const [minor, entry] of innerMap) {
-					yield { id: { minor }, root: entry };
+		for (const [major, innerRangeMap] of this.detachedNodeRangeMap) {
+			for (const rangeEntry of innerRangeMap) {
+				const id =
+					major !== undefined
+						? { major, minor: rangeEntry.start }
+						: { minor: rangeEntry.start };
+				for (let offset = 0; offset < rangeEntry.length; offset++) {
+					yield { id: { ...id, minor: id.minor + offset }, root: rangeEntry.value };
```

Comment:
There is a 1:1 relationship between `DetachedNodeId`s and `ForestRootId`s
```suggestion
					yield { id: { ...id, minor: id.minor + offset }, root: brand(rangeEntry.value + offset) };
```